test/test_mesh_pad_layer.cpp

```diff
@@ -0,0 +1,148 @@
+    using coord_t = double;
+    coord_t dx = coord_t(360.)/coord_t(nx - 1);
+    p_teca_variant_array_impl<coord_t> x = teca_variant_array_impl<coord_t>::New(nx);
+    coord_t *px = x->get();
+    for (unsigned long i = 0; i < nx; ++i)
+        px[i] = i*dx;
+
+    coord_t dy = coord_t(180.)/coord_t(ny - 1);
+    p_teca_variant_array_impl<coord_t> y = teca_variant_array_impl<coord_t>::New(ny);
+    coord_t *py = y->get();
+    for (unsigned long i = 0; i < ny; ++i)
+        py[i] = coord_t(-90.) + i*dy;
+
+    unsigned long nz = 1;
+    p_teca_variant_array_impl<coord_t> z = teca_variant_array_impl<coord_t>::New(nz);
+    z->set(0, 0.f);
+
+    p_teca_variant_array_impl<coord_t> t = teca_variant_array_impl<coord_t>::New(1);
+    t->set(0, 1.f);
+
+    unsigned long nxy = nx * ny;
+    p_teca_double_array ones_grid = teca_double_array::New(nxy);
+    double *p_ones_grid = ones_grid->get();
+
+    for (unsigned int i = 0; i < nxy; ++i)
+    {
+        p_ones_grid[i] = 1;
+    }
+
+    unsigned long wext[] = {0, nx - 1, 0, ny - 1, 0, 0};
+
+    p_teca_cartesian_mesh mesh = teca_cartesian_mesh::New();
+    mesh->set_x_coordinates("x", x);
+    mesh->set_y_coordinates("y", y);
+    mesh->set_z_coordinates("z", z);
+    mesh->set_whole_extent(wext);
+    mesh->set_extent(wext);
+    mesh->set_time(1.0);
+    mesh->set_time_step(0ul);
+    mesh->get_point_arrays()->append("ones_grid", ones_grid);
+
+    teca_metadata md;
+    md.set("whole_extent", wext, 6);
+    md.set("time_steps", std::vector<unsigned long>({0}));
+    md.set("variables", std::vector<std::string>({"ones_grid"}));
+    md.set("number_of_time_steps", 1);
+    md.set("index_initializer_key", std::string("number_of_time_steps"));
+    md.set("index_request_key", std::string("time_step"));
+
+    // build the pipeline
+    p_teca_dataset_source source = teca_dataset_source::New();
+    source->set_metadata(md);
+    source->set_dataset(mesh);
+
+    p_teca_mesh_padding mesh_padder = teca_mesh_padding::New();
+    mesh_padder->set_input_connection(source->get_output_port());
+    mesh_padder->set_px_low(px_low);
+    mesh_padder->set_px_high(px_high);
+    mesh_padder->set_py_low(py_low);
+    mesh_padder->set_py_high(py_high);
+
+    p_teca_mesh_layering mesh_layerer = teca_mesh_layering::New();
+    mesh_layerer->set_input_connection(mesh_padder->get_output_port());
+    mesh_layerer->set_n_layers(3);
+
+    //p_teca_dataset_capture transformed_o = teca_dataset_capture::New();
+    //transformed_o->set_input_connection(mesh_layerer->get_output_port());
+    //transformed_o->set_input_connection(mesh_padder->get_output_port());
+
+    p_teca_index_executive exe = teca_index_executive::New();
+    exe->set_start_index(0);
+    exe->set_end_index(0);
+
+    p_teca_cartesian_mesh_writer wri = teca_cartesian_mesh_writer::New();
+    wri->set_input_connection(mesh_layerer->get_output_port());
+    wri->set_executive(exe);
+    wri->set_file_name(out_file);
+
+    wri->update();
+
```

Comment:
please use the dataset diff to verify that the test succeeded, and consider using the Cartesian mesh source to generate the input.